### ISSUE TITLE
Use modern CMake syntax

### ISSUE THREAD

**Issue by vsoftco** [open-quantum-safe/liboqs#1494]
Instead of depending on `make/msbuild etc`, we should use only modern cmake and ctest, such as

```shell
cmake -B build -DCMAKE_FLAGS...
cmake --build build --parallel 4 [--target ]
ctest --test-dir build
```

This is platform-independent and saves a lot of headaches

**Comment by dstebila** [comment]
So `cmake --build build` would also replace ninja?

I don't mind, though I haven't had any trouble with our current build sequence.

The first two commands work fine for me, but `ctest --test-dir build` doesn't do anything.  I'm guessing this is because our test suite is under a custom target (`run_tests`) rather than some default name?

**Comment by vsoftco** [comment]
Yes, it'll replace ninja, and is completely platform-independent. I'll look into the test suite

**Comment by baentsch** [comment]
I like the platform-independence aspect (particularly looking at oqs-provider for Windows right now). What makes me wonder, though, is how this will impact build+test/CI runtime: `ninja` always nicely & automatically takes the maximum amount of CPUs available. The proposal above seems to require a hardcoded number of CPUs (that may or may not be available, i.e., run too slow if more CPUs were available and may overwhelm the machine if too many are spec'd).

**Comment by vsoftco** [comment]
@dstebila Thanks, that's good to know, I didn't know we can use the flag without a number

**Comment by SWilson4** [comment]
@Martyrshot took an initial stab at this in https://github.com/open-quantum-safe/liboqs/pull/1700. That PR would be a good jumping-off point for anyone who would like to take this up in the future.

**Comment by aidenfoxivey** [comment]
I believe that when you remove Ninja, then it just invokes make behind the scenes.

**Comment by aidenfoxivey** [comment]
<img width="2750" height="1902" alt="Image" src="https://github.com/user-attachments/assets/6fa09b52-85e8-4088-a414-91c4d10b6472" />

In this case, on a fresh VM, it just uses `gmake` when I invoke it with nothing else.

`cmake -B build`
`cmake --build build --parallel`

**Comment by dstebila** [comment]
Closing an old issue in the interests of clearing some backlog. In principle nothing prevents cmake experts from using `cmake -B build; cmake --build build`. The only thing lacking is use of `ctest` instead of our custom `run_tests` target, but at this point with limited dev resources I think this is low priority. If someone does want to take this on in the future, that's fine.
